tests/data/test_dataloader.py

```diff
@@ -429,27 +440,112 @@ def test_tokenize_example_malformed():
+            'role': 'assistant',
+            'content': 'Hi, User!'
+        }, {
+            'role': 'user',
+            'content': 'user message not followed by an assistant label'
+        }]
+    }
+    no_assistant_message = {
+        'messages': [{
+            'role': 'user',
+            'content': 'Hello GPT!'
+        }, {
+            'role': 'user',
+            'content': 'user message not followed by an assistant label'
+        }]
+    }
+    malformed_chat_examples = [
+        no_content, ends_with_user_role, no_assistant_message
+    ]
+    my_tokenizer = build_tokenizer('mosaicml/mpt-7b-chat', {})
+    for example in malformed_chat_examples:
+        with pytest.raises(Exception):
+            _tokenize_formatted_example(
+                example, my_tokenizer
+            )  # type: ignore (the typing here is supposed to be malformed)
+
+
+def test_tokenize_instruct_example_well_formed():
     tokenizer = transformers.AutoTokenizer.from_pretrained('gpt2')
 
     for prompt_key in _ALLOWED_PROMPT_KEYS:
         for response_key in _ALLOWED_RESPONSE_KEYS:
+
             example = {prompt_key: 'prompt', response_key: 'response'}
             tokenized_example = _tokenize_formatted_example(example, tokenizer)
             assert 'input_ids' in tokenized_example
             assert 'labels' in tokenized_example
 
 
+def test_tokenize_chat_example_well_formed():
+    chat_examples = [
+        {
+            'messages': [{
+                'role': 'user',
+                'content': 'Hello, GPT'
+            }, {
+                'role': 'assistant',
+                'content': 'this is my response'
+            }]
+        },  # prompt/response but in chat format
+        {
+            'messages': [
+                {
+                    'role': 'user',
+                    'content': 'Hello, GPT'
+                },
+                {
+                    'role': 'assistant',
+                    'content': 'this is my response'
+                },
+                {
+                    'role': 'user',
+                    'content': 'Nice to hear that.'
+                },
+                {
+                    'role': 'assistant',
+                    'content': 'multi-way chat works too!'
+                },
+            ]
+        },  # multi-way chat
+    ]
+
+    chat_tokenizer = build_tokenizer('mosaicml/mpt-7b-chat', {})
+    for chat_example in chat_examples:
+        last_message = chat_example['messages'][-1]['content']
+        earlier_messages = [
+            msg['content'] for msg in chat_example['messages'][:-1]
+        ]
+        prompt, response = _slice_chat_formatted_example(
+            chat_example, chat_tokenizer)
+        tokenized_example = _tokenize_formatted_example(chat_example,
+                                                        chat_tokenizer)
+        assert last_message in response
+        for earlier_message in earlier_messages:
+            assert earlier_message in prompt
```

Comment:
Can you do exact string matches here? Similar to test_tiktoken.py.

Reply:
feel free to pull it out into a smaller test that just tests the second chat_example. no need to do extract string comparison for both.